### DISCUSSION TITLE
Wanted to know how to set CI source.

### DISCUSSION THREAD

**Discussion by cb-krishnapatel** [danger/danger-js#1434] [Q&A]
I am using github actions, and i have a image which prepared from the dockerfile which is available here. The reason for this is to not build everytime the image when the action runs. and instead of that I'm directly using the image. when I'm running the image and passing the arguments and envs, its still failing to recognise CI source. The arguments and envs i took from the way github action run happens and from its logs. Can you please help here if I'm missing something. I have added image for the command and the error. 
![image](https://github.com/danger/danger-js/assets/124258547/4c074168-d275-44e8-b3b1-043e2f19052f)
<img width="899" alt="image" src="https://github.com/danger/danger-js/assets/124258547/b772e2b0-67df-4ecd-adb9-45ceb304d186">

**Comment by fbartho** [comment]
@cb-krishnapatel Looks like you missed an environment variable set by GitHub called `GITHUB_WORKFLOW`

https://github.com/danger/danger-js/blob/81645a6a7f37534d156ad495c639c914bf7053e1/source/ci_source/providers/GitHubActions.ts#L237

As an aside, it’s not obvious to me that you’ll save time if your docker image *only* has DangerJS. Now if your image is pulling other dependencies that take longer to install, then that might make sense!

**Reply by fbartho** [reply]
`ensureEnvKeysExists` requires that the value be defined to non-empty-string.

https://github.com/danger/danger-js/blob/81645a6a7f37534d156ad495c639c914bf7053e1/source/ci_source/ci_source_helpers.ts#L19

If you were capturing the values from a real pull request, it would have had the name of the workflow file, I'm not sure if it's actually the filename, or if it's the configured name inside that yaml.

It looks like you're running the docker image manually, rather than having Github Actions Workflows load the docker image for you? If Github did it, then it would already have the right environment variables, and they'd just be passed in for you?

Looking at the rest of this file, there seems to only be one other `GITHUB_` env var to look out for, `GITHUB_EVENT_PATH` (in addition to your flavor of `GITHUB_TOKEN` variable (there are a couple that Danger will read in different circumstances.

https://github.com/danger/danger-js/blob/81645a6a7f37534d156ad495c639c914bf7053e1/source/ci_source/providers/GitHubActions.ts

**Reply by cb-krishnapatel** [reply]
Thanks @fbartho ill have a look to this
